bin/init.sh

```diff
@@ -21,7 +21,7 @@ if [ ${ROOT} != "${GOPATH-$HOME/go}/src/istio.io/istio" ]; then
        exit 1
 fi
 
-which dep >/dev/null || go get -u github.com/golang/dep/cmd/dep
+which ${GOPATH}/bin/dep >/dev/null || go get -u github.com/golang/dep/cmd/dep
```

Comment:
you mean $GOROOT? $GOPATH has many paths.

Reply:
Thanks Kuat, GOPATH/bin/dep is being used everywhere, I guess we should change all of them

Reply:
Is there a problem with just `which dep` and installing it if necessary?

Reply:
The following is simpler and should be sufficient:
[ -f ${GOPATH}/bin/dep ] || go get -u github.com/golang/dep/cmd/dep

GOPATH won' t have many paths given the "export GOPATH=$TOP" that happens a few lines above.

Reply:
whoa, why do we override GOPATH? that seems counter to go practice. it's a standard practice to use system vendored libraries to make local changes instead of using dep.

Reply:
The top-level Makefile does it too.  It seems like a reasonable thing to do from a perspective of build hermetics and reproducibility.

Reply:
We can isolate the host OS with a container or a VM and use standard go practices.
Doing peculiar things with go tool seems to fall into the bazel trap of being overly opinionated about the build environments.

Reply:
I see GOPATH set in Makefile
# Make sure GOPATH is set based on the executing Makefile and workspace. *Will override
# GOPATH from the env.*
export GOPATH= $(shell cd ../../..; pwd)